compiler/rustc_codegen_ssa/src/codegen_attrs.rs

```diff
@@ -54,6 +54,12 @@ fn linkage_by_name(tcx: TyCtxt<'_>, def_id: LocalDefId, name: &str) -> Linkage {
 }
 
 fn codegen_fn_attrs(tcx: TyCtxt<'_>, did: LocalDefId) -> CodegenFnAttrs {
+    codegen_instance_attrs(tcx, InstanceKind::Item(did.to_def_id()))
+}
+
+fn codegen_instance_attrs(tcx: TyCtxt<'_>, instance_kind: InstanceKind<'_>) -> CodegenFnAttrs {
```

Comment:
Maybe `codegen_instance_attrs` could be a regular function that calls the `codegen_fn_attrs` query and drops the naked flag (and any other flags like no_mangle) as necessary depending on the instance kind?

Reply:
For this PR that is probably a nicer solution actually yeah.